src/storage/DataAccessorBasedStore.ts

```diff
@@ -0,0 +1,366 @@
+   * the identifier with differing trailing slash is requested.
+   * @param identifier - Identifier that needs to be checked.
+   */
+  protected async getNormalizedMetadata(identifier: ResourceIdentifier): Promise<RepresentationMetadata> {
+    const hasSlash = identifier.path.endsWith('/');
+    try {
+      return await this.accessor.getMetadata(identifier);
+    } catch (error: unknown) {
+      if (error instanceof NotFoundHttpError) {
+        return this.accessor.getMetadata(
+          { path: hasSlash ? trimTrailingSlashes(identifier.path) : ensureTrailingSlash(identifier.path) },
+        );
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Returns the result of `getNormalizedMetadata` or undefined if a 404 error is thrown.
+   */
+  protected async getSafeNormalizedMetadata(identifier: ResourceIdentifier):
+  Promise<RepresentationMetadata | undefined> {
+    try {
+      return await this.getNormalizedMetadata(identifier);
+    } catch (error: unknown) {
+      if (!(error instanceof NotFoundHttpError)) {
+        throw error;
+      }
+    }
+  }
+
+  /**
+   * Write the given resource to the DataAccessor. Metadata will be updated with necessary triples.
+   * In case of containers `handleContainerData` will be used to verify the data.
+   * @param identifier - Identifier of the resource.
+   * @param representation - Corresponding Representation.
+   * @param isContainer - Is the incoming resource a container?
+   * @param createContainers - Should parent containers (potentially) be created?
+   */
+  protected async writeData(identifier: ResourceIdentifier, representation: Representation, isContainer: boolean,
+    createContainers?: boolean): Promise<void> {
+    if (isContainer) {
+      await this.handleContainerData(representation);
+    }
+
+    if (createContainers) {
+      await this.createRecursiveContainers(await this.containerManager.getContainer(identifier));
+    }
+
+    const { metadata } = representation;
+    metadata.identifier = DataFactory.namedNode(identifier.path);
+    metadata.addQuads(this.metadataController.generateResourceQuads(metadata.identifier, isContainer));
+
+    if (isContainer) {
+      await this.accessor.writeContainer(identifier, representation.metadata);
+    } else {
+      await this.accessor.writeDataResource(identifier, representation.data, representation.metadata);
+    }
+  }
+
+  /**
+   * Verify if the incoming data for a container is valid (RDF and no containment triples).
+   * Adds the container data to its metadata afterwards.
+   *
+   * @param representation - Container representation.
+   */
+  protected async handleContainerData(representation: Representation): Promise<void> {
+    let quads: Quad[];
+    try {
+      quads = await this.metadataController.parseQuads(representation.data);
+    } catch (error: unknown) {
+      // Make error more readable to know what happened
+      if (error instanceof Error) {
```

Comment:
Shouldn't that always be the case?

Reply:
In short: probably, but typing wise there is no way to guarantee.

This is related to the linter changes that were added some time ago. Since the latest Typescript version error that you catch can be typed in 2 ways: as `any` or `unknown`. The linter forces us to use `unknown` since there is no way to know what we are actually catching (as `throw "justastring"` is valid Javascript). In practice this is probably always going to be an Error object for us, unless we're using an external library that throws non-Error things (I'm assuming here this never happens in core Node libraries). But so typing ways there is no way to guarantee this since you can't have different catches for different types of errors (such as in Java).